cmd/object-handlers-common.go

```diff
@@ -298,6 +306,12 @@ func checkPreconditions(ctx context.Context, w http.ResponseWriter, r *http.Requ
 			writeHeadersPrecondition(w, objInfo)
 			writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrPreconditionFailed), r.URL)
 			return true
+		} else if !checkIfUnmodifiedSince(r, objInfo) {
```

Comment:
If do this
1. `If-Match` is true and `If-Unmodified-Since` is false, return 200.
2. `If-Match` is true and `If-Unmodified-Since` is true, return 200.
3. `If-Match` is true, return 200.
4. `If-Unmodified-Since` is true, return 200.
5. `If-Unmodified-Since` is false, return 412.

I think there's no problem with doing this, Thanks!